File: src/main/resources/templates/go/solution/context.go
package ${packageName}

import "time"

type Context struct {
	dates         []time.Time
	sortAlgorithm SortStrategy
}

func NewContext() *Context {
	return new(Context)
}

func (c *Context) GetDates() []time.Time {
	return c.dates
}

func (c *Context) SetDates(dates []time.Time) {
	c.dates = dates
}

func (c *Context) GetSortAlgorithm() SortStrategy {
	return c.sortAlgorithm
}

func (c *Context) SetSortAlgorithm(strategy SortStrategy) {
	c.sortAlgorithm = strategy
}

// Sort runs the configured sort algorithm.
func (c *Context) Sort() {
	if c.sortAlgorithm == nil {
		panic("sortAlgorithm has not been set")
	}

	if c.dates == nil {
		panic("dates has not been set")
	}

	c.sortAlgorithm.PerformSort(c.dates)
}

File: src/main/resources/templates/ruby/solution/src/context.rb
# frozen_string_literal: true

class Context
  attr_accessor :dates, :sort_algorithm

  # Runs the configured sort algorithm.
  def sort
    raise "sort_algorithm not set" if @sort_algorithm.nil?
    raise "dates not set" if @dates.nil?

    @sort_algorithm.perform_sort(@dates)
  end
end
